pkg/storage/apis/app/v1beta2/fybrikstorageaccount_types.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1beta2
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"fybrik.io/fybrik/pkg/model/taxonomy"
+	"fybrik.io/fybrik/pkg/serde"
+)
+
+const typeKey = "type"
+const idKey = "id"
+const geographyKey = "geography"
+const regionKey = "region"
+const secretRefKey = "secretRef"
+
+// FybrikStorageAccountSpec defines the desired state of FybrikStorageAccount
+// +kubebuilder:pruning:PreserveUnknownFields
+type FybrikStorageAccountSpec struct {
+	// Identification of a storage account
+	// +required
+	ID string `json:"id"`
+	// +required
+	// A name of k8s secret deployed in the control plane.
+	SecretRef string `json:"secretRef"`
+	// +required
+	// Type of the storage, e.g., s3
+	Type taxonomy.ConnectionType `json:"type"`
+	// +required
+	// Storage geography
+	Geography taxonomy.ProcessingLocation `json:"geography"`
+	// Additional storage properties, specific to the storage type
+	AdditionalProperties serde.Properties `json:"-"`
+}
+
+// FybrikStorageAccountStatus defines the observed state of FybrikStorageAccount
+type FybrikStorageAccountStatus struct {
+}
+
+// FybrikStorageAccount defines a storage account used for copying data.
+// It contains connection details of the shared storage and refers to the secret that stores storage credentials.
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+type FybrikStorageAccount struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +required
+	Spec   FybrikStorageAccountSpec   `json:"spec"`
+	Status FybrikStorageAccountStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// FybrikStorageAccountList contains a list of FybrikStorageAccount
+type FybrikStorageAccountList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []FybrikStorageAccount `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&FybrikStorageAccount{}, &FybrikStorageAccountList{})
+}
+
+func (o FybrikStorageAccountSpec) MarshalJSON() ([]byte, error) {
+	toSerialize := map[string]interface{}{
+		idKey:        o.ID,
+		secretRefKey: o.SecretRef,
+		typeKey:      o.Type,
+		geographyKey: o.Geography,
+	}
+	for key, value := range o.AdditionalProperties.Items {
+		toSerialize[key] = value
+	}
+
+	return json.Marshal(toSerialize)
+}
+
+func (o *FybrikStorageAccountSpec) UnmarshalJSON(bytes []byte) (err error) {
+	items := make(map[string]interface{})
+	if err = json.Unmarshal(bytes, &items); err == nil {
+		o.ID = items[idKey].(string)
+		o.SecretRef = items[secretRefKey].(string)
+		delete(items, idKey)
+		delete(items, secretRefKey)
+		typeVal, exists := items[typeKey]
+		if !exists {
+			// conversion from v1beta1 is required:
```

Comment:
Should the backward compatibility to v1beta1 be done here?  Shouldn't this be handled by a webhook or some other mechanism addressing backward compatibility?

Reply:
What was decided about this @shlomitk1 @roytman ?  I see you took the code out, but I don't see the handling of the backward compatibility anyplace else.

Reply:
It still happens during the object construction. Webhooks can come on top of that, but a default conversion is required if no webhooks are running.

Reply:
Done. Served is set to true because otherwise deprecated warnings are not displayed.